packages/dataviews/src/pagination.js

```diff
@@ -27,36 +27,33 @@ const Pagination = memo( function Pagination( {
 				justify="end"
 				className="dataviews-pagination"
 			>
-				<HStack justify="flex-start" expanded={ false } spacing={ 2 }>
-					{ createInterpolateElement(
-						sprintf(
+				<HStack justify="flex-start" expanded={ false } spacing={ 1 }>
+					<SelectControl
+						aria-label={ __( 'Current page' ) }
+						value={ view.page }
+						className="dataviews-pagination__page-selector"
+						options={ Array.from( Array( totalPages ) ).map(
+							( _, i ) => {
+								const page = i + 1;
+								return { value: page, label: page };
+							}
+						) }
+						onChange={ ( newValue ) => {
+							onChangeView( {
+								...view,
+								page: +newValue,
+							} );
+						} }
+						size={ 'compact' }
+						__nextHasNoMarginBottom
+					/>
+					<span>
+						{ sprintf(
 							// translators: %s: Total number of pages.
-							_x( 'Page <CurrenPageControl /> of %s', 'paging' ),
+							'of %s',
```

Comment:
I'm not sure about how easy it is for screen readers to parse the information (ie. `Current page is X of Y`).

The risk with splitting the strings like this is that the "of Y" structure may not hold true for all locales.

cc'ing @andrewhayward in case he can help with this